builtin/logical/consul/path_roles.go

```diff
@@ -37,8 +37,14 @@ for 'client' tokens. Required for Consul pre-1.4.`,
 
 			"policies": {
 				Type: framework.TypeCommaStringSlice,
-				Description: `List of policies to attach to the token. Required
-for Consul 1.4 or above.`,
+				Description: `List of policies to attach to the token. Either policies or 
+roles are required for Consul 1.5 and above, or just policies if using Consul 1.4.`,
+			},
+
+			"roles": {
```

Comment:
Similar to `consul_namespace`, I wonder if we should name this `consul_roles` so that the command feels a bit more explicit when specifying this.

```sh
$ vault write consul/roles/my-role roles=api-server
```

vs 

```sh
$ vault write consul/roles/my-role consul_roles=api-server
```

Reply:
Good catch. I meant to include a comment myself (but forgot...) to start this discussion.
I initially considered consul_roles, but used just roles to begin because there's no -role flag like there is with -namespace. That makes it a bit more separate than namespaces were between the two. However when looking at these side-by-side, it likely is better to use consul_roles to be certain there's no ambiguity down the line.

Reply:
Changed here https://github.com/hashicorp/vault/pull/14014/commits/aa1901b0cb5e6f64ad77cd8f807b2b51195ac534

Reply:
@calvn @robmonte  Question, is the parameter naming scheme we are using here meant to disambiguate consul roles from vault roles? If so that seems to be inconsistent with other parameters like: [policies](https://www.vaultproject.io/api-docs/secret/consul#policies). I wonder if we should rename `policies` etc. to match what we are doing for `roles`.

Reply:
Yes, that was the goal. Namespaces look particularly confusing if we just used the word "namespace" like so:
`vault read -namespace="ns1" consul/creds/my-role namespace="ns2"`
`vault read -namespace="ns1" consul/creds/my-role consul_namespace="ns2"`
While Consul's roles aren't quite as confusing as that, I think it still helps avoid confusion to prefix "consul_" to it as well.

As for policies, I'm not sure if it would be better to change that after it's been in-use for such a long time.

Reply:
Thanks @robmonte !